### DISCUSSION TITLE
about  liqo's  usage ...

### DISCUSSION THREAD

**Discussion by whybeyoung** [liqotech/liqo#1222] [Q&A] [answered]
HI, all:
   Very excited to see the job that liqo did. After tried it , and i have some questions... The demo nginx pod is very simple. I have a 
MLOPS case:
   you have two k8s clusters now: A, B， ClusterA's duty  is keep the HA  of the control plane: such as argo server,argo controllers,volucano schedulers...etc which are not  control plane of k8s itself.  And Cluster B's duty is pure  trainning cluster. 
  generally  we always do the HA for k8s apiserver and shceduler,etcd...   We even use the kube-on-kube approach to ensure high availability of the masters.  Howerver , except k8s's HA, we need argo HA, volcanon HA, and we don't want to waste machine resources. we have multiclusters. We want to divide multiple clusters into multiple sub-clusters by usages, and use methods such as liqo to ensure the high availability of the control plane of the business layer in only one cluster.  

   I have a small try:

<img width="1166" alt="image" src="https://user-images.githubusercontent.com/10629930/166220170-9ba86d82-9e70-4063-a0a6-99c047f47730.png">

In briefly: 

*  Not very sure that  is the liqo suitable for our case...
*  i'm very confused about  how will  the argo server/controller in clusterA to manage  the clusterB's workflows if using liqo....
*  how to do the business layer such as argo controller's rbac offloading   if using liqo..

Thanks.

**Comment by giorio94** [comment, marked as answer]
Hi @berlinsaint,

As far as I understood, you would like to spread the control-plane pods of certain components (e.g., argo) across multiple clusters, to achieve high availability. Am I correct?

Liqo v0.4.0 does not support offloading pods that need to interact with the Kubernetes API server, which I believe is the reason you are getting that forbidden error. This is a feature we are planning to introduce in the next release (actually, an initial version is already merged upstream, and you can use it configuring the liqo version flag equal to the sha of the latest commit on master during the installation process), to allow offloaded pods to interact with the API server of the origin cluster (not that where it is running in that moment). As for RBACs, it would get the same privileges it would have if executed locally. Would this fit your use-case?

**Reply by whybeyoung** [reply]
thanks buddy ，I will have a try
